bugwarrior/services/trello.py

```diff
@@ -0,0 +1,198 @@
+            ((c['memberCreator']['username'], c['data']['text']) for c in comments),
+            card_json["shortUrl"])
+        return annotations
+
+
+    def get_boards(self):
+        """
+        Get the list of boards to pull cards from.  If the user gave a value to
+        trello.include_boards use that, otherwise ask the Trello API for the
+        user's boards.
+        """
+        if self.config_has('include_boards'):
+            for boardid in self.config_get('include_boards', aslist):
+                # Get the board name
+                yield self.api_request(
+                    "/1/boards/{id}".format(id=boardid), fields='name')
+        else:
+            boards = self.api_request("/1/members/me/boards", fields='name')
+            for board in boards:
+                yield board
+
+    def get_lists(self, board):
+        """
+        Returns a list of the filtered lists for the given board
+        This filters the trello lists according to the configuration values of
+        trello.include_lists and trello.exclude_lists.
+        """
+        lists = self.api_request(
+            "/1/boards/{board_id}/lists/open".format(board_id=board),
+            fields='name')
+        try:
+            include_lists = self.config_get('include_lists', aslist)
+            lists = [l for l in lists if l['name'] in include_lists]
+        except NoOptionError:
+            pass
+        try:
+            exclude_lists = self.config_get('exclude_lists', aslist)
+            lists = [l for l in lists if l['name'] not in exclude_lists]
+        except NoOptionError:
+            pass
+        return lists
+
+    def get_cards(self, list_id):
+        """ Returns an iterator for the cards in a given list, filtered
+        according to configuration values of trello.only_if_assigned and
+        trello.also_unassigned """
+        params = {'fields': 'name,idShort,shortLink,shortUrl,url,labels'}
+        member = self.config_get_default('only_if_assigned', None)
+        unassigned = self.config_get_default('also_unassigned', False, asbool)
+        if member is not None:
+            params['members'] = 'true'
+            params['member_fields'] = 'username'
+        cards = self.api_request(
+            "/1/lists/{list_id}/cards/open".format(list_id=list_id),
+            **params)
+        for card in cards:
+            if (member is None
+                    or member in [m['username'] for m in card['members']]
+                    or (unassigned and not card['members'])):
+                yield card
+
+    def get_comments(self, card_id):
+        """ Returns an iterator for the comments on a certain card. """
+        params = {'filter': 'commentCard', 'memberCreator_fields': 'username'}
+        comments = self.api_request(
+            "/1/cards/{card_id}/actions".format(card_id=card_id),
+            **params)
+        for comment in comments:
+            assert comment['type'] == 'commentCard'
```

Comment:
I first put it there for my own sake during development but then I figured it didn't hurt to leave it there and it could help diagnose some unexpected API behavior from trello.